microsoft/testsuites/performance/storageperf.py

```diff
@@ -125,6 +143,195 @@ def perf_premium_datadisks_io(self, node: Node, environment: Environment) -> Non
+            ),
+            network_interface=Synthetic(),
+        ),
+    )
+    def perf_storage_over_nfs_synthetic_udp_4k(self, environment: Environment) -> None:
+        self._perf_nfs(
+            environment,
+            protocol="udp",
+        )
+
+    def _perf_nfs(
+        self,
+        environment: Environment,
+        server_raid_disk_name: str = "/dev/md0",
+        server_raid_disk_mount_dir: str = "/mnt/nfs_share",
+        client_nfs_mount_dir: str = "/mnt/nfs_client_share",
+        protocol: str = "tcp",
+        filename: str = "fiodata",
+        block_size: int = 4,
+        start_iodepth: int = 1,
+        max_iodepth: int = 1024,
+    ) -> None:
+        server_node = cast(RemoteNode, environment.nodes[0])
+        client_node = cast(RemoteNode, environment.nodes[1])
+
+        # get testname from stack
+        test_case_name = inspect.stack()[1][3]
+
+        # Run test only on Debian, SLES and Redhat distributions
+        if (
+            not isinstance(server_node.os, Redhat)
+            and not isinstance(server_node.os, Debian)
+            and not isinstance(server_node.os, SLES)
+        ):
+            raise SkippedException(f"{server_node.os} not supported")
+
+        # calculate the number of jobs to run. The max number of jobs is equal
+        # to the core count of the node.
+        num_jobs = []
+        core_count = client_node.tools[Lscpu].get_core_count()
+        for i in range(start_iodepth, max_iodepth + 1):
+            num_jobs.append(min(2 ** (i - 1), core_count))
+
+        # setup raid on server
+        server_data_disks = server_node.features[Disk].get_raw_data_disks()
+        server_data_disk_count = len(server_data_disks)
+        server_partition_disks = _format_disk(server_node, server_data_disks)
+        _stop_raid(server_node)
+        _make_raid(server_node, server_partition_disks)
+
+        # mount raid disk on server
+        server_node.shell.mkdir(
+            PurePosixPath(server_raid_disk_mount_dir), exist_ok=True
+        )
+        server_node.tools[Mkfs].format_disk(server_raid_disk_name, FileSystem.ext4)
+        server_node.tools[Mount].mount(
+            server_raid_disk_name, server_raid_disk_mount_dir, options="nobarrier"
+        )
+
+        # setup nfs on server
+        server_node.tools[NFSServer].setup(
+            [client_node.public_address], server_raid_disk_mount_dir
+        )
+
+        # setup raid on client
+        client_node.tools[NFSClient].setup(
+            server_node.public_address,
+            server_raid_disk_mount_dir,
+            client_nfs_mount_dir,
+            protocol,
+        )
+
+        # Run a dummy fio job on client to create required test files and
+        # transfer over network. If this step is skipped, it can impact
+        # the test result.
+        client_node.tools[Fio].launch(
+            name="prepare",
+            filename=filename,
+            mode=FIOMODES.read.name,
+            time=120,
```

Comment:
check if smaller value works.

Reply:
It needs around 1000 seconds time to run. So I set it to 1200 to give it some more slack.

Reply:
What the "120" mean here? 1200 seconds? The unit is confusing.

Reply:
I think I confused fio timeout with test timeout. The timeout here is "120" second for the fio command. It was taken from v2. I have also observed on ubuntu that it takes appoximately 100 seconds to run it, so 120 sec seems like a good value.

Reply:
The default value is 120 too, it doesn't need to specify it here. It's the same for other values, if the default value is used, avoid to specify it to keep it clear.